### PR TITLE
Equatables for Result and FrisbeeError

### PR DESCRIPTION
**Pull request by williamhjcho**
- Moved `Equatable` extensions from tests to framework
- Moved `Result` `get-properties` extensions to framework
- Add rest of cases on `FrisbeeError: Equatable`
- Add `==` overload when `Entity: Equatable` on `Result`
- Add `Sequence.extensiveCombine` on testing target to enable cartesian product of `Sequences`
    `[0,1] x [a, b, c] = [(0,a), (0,b), (0,c), (1,a), (1,b), (1,c)]`

### CODE REVIEW

**Comment by ronanrodrigo** [comment]
These equatables are just use in test target. Do you think it is necessary to move them to production target?

**Comment by williamhjcho** [comment]
Makes it easier for the future, there is also no need for them to **not** be `equatable`

File: Tests/FrisbeeTests/Entities/FrisbeeErrorTests.swift [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+                XCTAssertEqual(lhs.element, rhs.element)
+            } else {
+                XCTAssertNotEqual(lhs.element, rhs.element)
+            }
+        }
+    }
+}
+
+extension FrisbeeError {
+    static func all(error: Error) -> [FrisbeeError] {
```
Comment (ronanrodrigo) [inline Tests/FrisbeeTests/Entities/FrisbeeErrorTests.swift:28]: To simplify this function, maybe `error` could be just some string. And maybe `SomeError` doesn't need to exist 🙂

**Review by ronanrodrigo — COMMENTED** [review]
Nice @william-wc 👏

**Pull request merged**